portfolio/src/main/java/com/google/sps/data/BookResponseParser.java

```diff
@@ -0,0 +1,177 @@
+        builder.publishedDate(tempPubDate);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(PUBLISHER)) {
+        tempPublisher = jsonObject.getJSONObject(VOLUME_INFO).get(PUBLISHER).toString();
+        builder.publisher(tempPublisher);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(MATURITY_RATING)) {
+        tempMatRate = jsonObject.getJSONObject(VOLUME_INFO).get(MATURITY_RATING).toString();
+        builder.maturityRating(tempMatRate);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(CATEGORIES)) {
+        tempCategories =
+            jsonArrayToStringArray(jsonObject.getJSONObject(VOLUME_INFO).getJSONArray(CATEGORIES));
+        builder.categories(tempCategories);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(AUTHORS)) {
+        tempAuthors =
+            jsonArrayToStringArray(jsonObject.getJSONObject(VOLUME_INFO).getJSONArray(AUTHORS));
+        builder.authors(tempAuthors);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(IMAGE_LINKS)
+          && jsonObject.getJSONObject(VOLUME_INFO)
+                 .getJSONObject(IMAGE_LINKS)
+                 .has(THUMBNAIL)) { // if imageLinks and the thumbnail are present
+        tempThumbnail = jsonObject.getJSONObject(VOLUME_INFO)
+                            .getJSONObject(IMAGE_LINKS)
+                            .get(THUMBNAIL)
+                            .toString();
+        builder.thumbnail(tempThumbnail);
+      }
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+
+    Book book = builder.build();
+    return book;
+  }
+
+  private static ArrayList<String> jsonArrayToStringArray(JSONArray jsArray) {
+    ArrayList<String> strArray = new ArrayList<String>();
+
+    for (int j = 0; j < jsArray.length(); j++) {
+      try {
+        strArray.add(jsArray.get(j).toString());
+
+      } catch (JSONException e) {
+        e.printStackTrace();
+      }
+    }
+    return strArray;
+  }
+
+  private static Map<String, String> getIsbn(JSONObject item) throws JSONException {
+    Gson gson = new Gson();
+
+    String tempType = "", tempISBN = "";
+    Map<String, String> isbnMap = new HashMap<String, String>();
+    int counter = 0;
+    if (item.getJSONObject(VOLUME_INFO).has("industryIdentifiers")) {
+      JSONArray tempArr = item.getJSONObject(VOLUME_INFO).getJSONArray("industryIdentifiers");
+      Type listType = new TypeToken<List<HashMap<String, String>>>() {}.getType();
+      List<HashMap<String, String>> isbns = gson.fromJson(tempArr.toString(), listType);
+
+      for (HashMap<String, String> map : isbns) {
+        counter = 0;
+        for (HashMap.Entry<String, String> entry : map.entrySet()) {
+          if (counter == 0) {
+            tempISBN = entry.getValue(); // Key is identifier
+          }
+          if (counter == 1) {
+            tempType = entry.getValue();
+            // Key is type
+          }
+          counter++;
+        }
```

Comment:
When iterating over hashmap, the order of keys is not guaranteed. From [javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html), `This class makes no guarantees as to the order of the map; in particular, it does not guarantee that the order will remain constant over time `.

If the order changes, this logic will break. I think better way to implement this would be by comparing the key:
```
        for (HashMap.Entry<String, String> entry : map.entrySet()) {
          if (entry.getKey().equals("identifier")) {
            tempISBN = entry.getValue();
          } else if (entry.getKey().equals("type")) {
            tempType = entry.getValue();
          }
        }
```

And, since, you already know the keys in the hashmap, an even better way is:

```        
tempISBN = map.get("identifier");
tempType = map.get("type");
```